File: backend-python/app/modules/system/dto/auth_request.py
from pydantic import BaseModel, Field


class LoginRequest(BaseModel):
    username: str
    password: str
    device_type: str = Field(alias="deviceType")


class PasswordRequest(BaseModel):
    old_password: str = Field(alias="oldPassword")
    new_password: str = Field(alias="newPassword")


class ProfileUpdateRequest(BaseModel):
    real_name: str = Field(alias="realName")
    phone: str | None = None
    email: str | None = None

File: backend-rust/src/modules/system/dto/auth_request.rs
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LoginRequest {
    pub username: String,
    pub password: String,
    pub device_type: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PasswordRequest {
    pub old_password: String,
    pub new_password: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProfileUpdateRequest {
    pub real_name: String,
    pub phone: Option<String>,
    pub email: Option<String>,
}
